src/apis/emscripten/jmp.rs

```diff
@@ -0,0 +1,25 @@
+use crate::webassembly::Instance;
+use libc::{c_int, c_void};
+
+/// setjmp
+pub extern "C" fn __setjmp(env_addr: u32, instance: &mut Instance) -> c_int {
```

Comment:
Exposing setjmp and longjmp to wasm is extremely unsafe. We need to find a safer way of doing this. Off the bat, we could generate opaque IDs and store the jumpbuf data outside of the wasm memory. So, when they call longjmp, we look up the right place to jump.

Reply:
You are completely right. I've just pushed a fix for this.

Reply:
We will use a external `jumps` field on the `emscripten_data`. The opaque IDs will effectively the index of the jump buffer in the `jumps` field.